In [40]:
import pandas as pd
import numpy as np
import re
import nltk
import math
from nltk.tokenize import word_tokenize

In [41]:
df = pd.read_csv('commentary_686.csv')

In [42]:
df

srno                                               Data
0       0  8.55am Good morning and welcome to our coverag...
1       1  Bracken to Smith, 1 run uppish and loose pushe...
2       2  Bracken to de Villiers, no run wide of off stu...
3       3  Bracken to de Villiers, no run again that swun...
4       4  Bracken to de Villiers, no run a similar line,...
..    ...                                                ...
596   596  A clinical performance from the defending cham...
597   597  Australia win by seven wickets The South Afric...
598   598  That's all from Jenny Thompson, Will Luke and ...
599   599  Australia won by 7 wickets (with 111 balls rem...
600   600          Player of the match is Glenn McGrath, AUS

[601 rows x 2 columns]

In [43]:
corpus = ' '.join(df['Data'])

In [44]:
corpus

"8.55am Good morning and welcome to our coverage of the second semi-final between Australia and South Africa, with the winners taking on Sri Lanka in Saturday's final. Team and toss news to follow any second ... Bracken to Smith, 1 run uppish and loose pushed drive, in the air but into space, third man fields Bracken to de Villiers, no run wide of off stump, plenty of swing, aided by a stiff breeze, and de Villiers leaves Bracken to de Villiers, no run again that swung into the right-hander, de Villiers into line and drops into the leg side Bracken to de Villiers, no run a similar line, de Villiers pushes the ball no more than five or six yards into the leg side Bracken to de Villiers, no run outside off and wide enough to leave Bracken to de Villiers, FOUR runs de Villiers pushes forward, a thick edge shoots along the ground to the right of solitary slip and the ball flies across the outfield McGrath to Smith, 1 run Smith whips that from middle-and-leg into the on side, an easy run to

In [45]:
# def clean_text(corpus):
article=corpus.split(".")
article


['8',
 "55am Good morning and welcome to our coverage of the second semi-final between Australia and South Africa, with the winners taking on Sri Lanka in Saturday's final",
 ' Team and toss news to follow any second ',
 '',
 '',
 ' Bracken to Smith, 1 run uppish and loose pushed drive, in the air but into space, third man fields Bracken to de Villiers, no run wide of off stump, plenty of swing, aided by a stiff breeze, and de Villiers leaves Bracken to de Villiers, no run again that swung into the right-hander, de Villiers into line and drops into the leg side Bracken to de Villiers, no run a similar line, de Villiers pushes the ball no more than five or six yards into the leg side Bracken to de Villiers, no run outside off and wide enough to leave Bracken to de Villiers, FOUR runs de Villiers pushes forward, a thick edge shoots along the ground to the right of solitary slip and the ball flies across the outfield McGrath to Smith, 1 run Smith whips that from middle-and-leg into the on

In [46]:
sentences = []
for sentence in article :
    sentence = re.sub('[^a-zA-Z]',' ',str(sentence))
    sentence= re.sub(r'\s+',' ',sentence)
    sentences.append(sentence)
sentences.pop()
display = " ".join(sentences)
print('intial text: ')
print(display)
print('\n')
print(sentences)
filtered_list = [item for item in sentences if item.strip()]

# Print the filtered list
print(filtered_list)

intial text: 
   am Good morning and welcome to our coverage of the second semi final between Australia and South Africa with the winners taking on Sri Lanka in Saturday s final  Team and toss news to follow any second     Bracken to Smith run uppish and loose pushed drive in the air but into space third man fields Bracken to de Villiers no run wide of off stump plenty of swing aided by a stiff breeze and de Villiers leaves Bracken to de Villiers no run again that swung into the right hander de Villiers into line and drops into the leg side Bracken to de Villiers no run a similar line de Villiers pushes the ball no more than five or six yards into the leg side Bracken to de Villiers no run outside off and wide enough to leave Bracken to de Villiers FOUR runs de Villiers pushes forward a thick edge shoots along the ground to the right of solitary slip and the ball flies across the outfield McGrath to Smith run Smith whips that from middle and leg into the on side an easy run to the man 

In [ ]:
import torch
from transformers import AutoTokenizer
from scipy.special import softmax
from transformers import AutoModelForSequenceClassification

c:\Users\kulu1\AppData\Local\Programs\Python\Python312\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [ ]:
Model = f"cardiffnlp/twitter-roberta-base-sentiment"
tokenizer = AutoTokenizer.from_pretrained(Model)
model = AutoModelForSequenceClassification.from_pretrained(Model)

In [ ]:
text = df['Data'][0]

# Encode the text
encoded_text = tokenizer(text, return_tensors='pt')

# Print the encoded text
print(encoded_text)

{'input_ids': tensor([[    0,   398,     4,  3118,   424,  2497,   662,     8,  2814,     7,
            84,  1953,     9,     5,   200,  4126,    12,  6156,   227,  1221,
             8,   391,  1327,     6,    19,     5,  4251,   602,    15,  6604,
          8500,    11,   378,    18,   507,     4,  2711,     8, 13027,   340,
             7,  1407,   143,   200,  1666,     2]]), 'attention_mask': tensor([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
         1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]])}


The output is theresult of encoding text data using a pre-trained tokenizer, likely from the Hugging Face Transformers library. Let's break down the components of the encoded text:

input_ids: This tensor represents the input sequence encoded as integer indices. Each integer corresponds to a token in the vocabulary of the tokenizer. In this case, the sequence seems to consist of integer indices representing individual tokens.Example: tensor([[0, 398, 4, ..., 200, 1666, 2]])
attention_mask: This tensor indicates which tokens the model should pay attention to (1) and which tokens it should ignore (0). It ensures that the model does not attend to padding tokens. In this case, all tokens seem to be attended to, as indicated by the value 1.Example: tensor([[1, 1, 1, ..., 1, 1, 1]])
These tensors represent the input data prepared for feeding into a neural network model for natural language processing tasks, such as classification or generation. The input_ids tensor contains the tokenized input sequence, while the attention_mask tensor ensures that the model focuses only on relevant tokens.

In [ ]:
def polarity_scores_roberta(example):
    encoded_text = tokenizer(example, return_tensors='pt')
    output = model(**encoded_text)
    score = output[0][0].detach().numpy()
    score = softmax(score)
    scores_dict = {
    "roberta-neg":score[0],
    "roberta-neu":score[1],
    "roberta-pos":score[2]
    }
    return scores_dict

In [ ]:
from tqdm import tqdm
res = {}
df['ID'] = range(1, len(df) + 1)
for i, row in tqdm(df.iterrows(), total=len(df)):
    text = str(row["Data"])
    matchno = row["ID"]

    roberta_result = polarity_scores_roberta(text)

    # Combine results
    both = { **roberta_result}
    res[matchno] = both

100%|██████████| 601/601 [01:00<00:00,  9.98it/s]


In [ ]:
res

In [ ]:
ball_ball = pd.DataFrame.from_dict(res,orient="index")
ball_ball

roberta-neg  roberta-neu  roberta-pos
1       0.003172     0.554784     0.442044
2       0.027088     0.911273     0.061639
3       0.010638     0.792396     0.196966
4       0.032006     0.921737     0.046258
5       0.058220     0.891869     0.049911
..           ...          ...          ...
597     0.384558     0.468084     0.147358
598     0.018247     0.534644     0.447108
599     0.002051     0.046878     0.951071
600     0.025116     0.786292     0.188592
601     0.011369     0.845010     0.143621

[601 rows x 3 columns]

In [ ]:
ballers = []
batsmen=[]
df['ID'] = range(1, len(df) + 1)
for i, row in tqdm(df.iterrows(), total=len(df)):
    text = str(row["Data"])
    matchno = row["ID"]

    baller = ollama("who is the baller  in give me just the name"+text)
    batsman = ollama("who is thebatsman in ,give me just the name"+text)
    if baller not in ballers :
        ballers.append(baller)
    if batsman not in batsmen :
        batsmen.append(batsman)

print(ballers)
print(batsmen)


In [66]:
import spacy 
import spacy.cli

In [67]:
spacy.cli.download("en_core_web_sm")

✔ Download and installation successful
You can now load the package via spacy.load('en_core_web_sm')
⚠ Restart to reload dependencies
If you are in a Jupyter or Colab notebook, you may need to restart Python in
order to load all the package's dependencies. You can do this by selecting the
'Restart kernel' or 'Restart runtime' option.


In [68]:
sp_sm = spacy.load('en_core_web_sm')

In [73]:
def spacy_large_ner(document):
  print({(ent.text.strip(), ent.label_) for ent in sp_sm(document).ents})


In [ ]:
# ballers = []
# batsmen=[]
df['ID'] = range(1, len(df) + 1)
for i, row in tqdm(df.iterrows(), total=len(df)):
    text = str(row["Data"])
    matchno = row["ID"]
    spacy_large_ner(text)
   

In [75]:
df

srno                                               Data   ID
0       0  8.55am Good morning and welcome to our coverag...    1
1       1  Bracken to Smith, 1 run uppish and loose pushe...    2
2       2  Bracken to de Villiers, no run wide of off stu...    3
3       3  Bracken to de Villiers, no run again that swun...    4
4       4  Bracken to de Villiers, no run a similar line,...    5
..    ...                                                ...  ...
596   596  A clinical performance from the defending cham...  597
597   597  Australia win by seven wickets The South Afric...  598
598   598  That's all from Jenny Thompson, Will Luke and ...  599
599   599  Australia won by 7 wickets (with 111 balls rem...  600
600   600          Player of the match is Glenn McGrath, AUS  601

[601 rows x 3 columns]

In [80]:

def getplayers(document):
    df['ID'] = range(1, len(df) + 1)
    print(ollama("give me the name of the players from the commentary and nothing else "+document))
    
getplayers(corpus)

Australia vs South Africa, World Cup Final: Australia win by 7 wickets

Australia had a dominant performance in the World Cup final against South Africa, winning by 7 wickets with 111 balls remaining. The Australians were in command from the start, restricting South Africa to a low score and then chasing down the target with ease.

The Australian bowling attack was on top of their game, with Glenn McGrath taking 3 wickets for 25 runs in his 10 overs. Brett Lee and Jason Gillespie also took 2 wickets each, and the South African batsmen found it difficult to score runs against them.

In response, Australia got off to a strong start with the bat, with openers Justin Langer and Matthew Hayden putting on 50 runs in the first 10 overs. Hayden was the more aggressive of the two, hitting 3 fours and a six in his 40-ball stay at the crease. Langer was more circumspect, but still managed to score at a healthy rate.

When Hayden was dismissed for 27 runs, Australia were 50 runs ahead of South Afr

In [85]:
player_pattern = r"(?<=to\s)\b[A-Z][a-zA-Z\s]+\b"

# Extract player names from commentary
player_names = re.findall(player_pattern, corpus)

print(player_names)

['Smith', 'Smith', 'Smith', 'Smith', 'Smith', 'Kallis', 'Kallis', 'Kallis', 'Kallis', 'Kallis', 'Kallis', 'Kallis', 'Kallis', 'Kallis', 'Gibbs', 'Gibbs', 'Hogg at mid', 'Gibbs', 'Gibbs', 'Gibbs', 'Gilchrist', 'Prince', 'Gibbs', 'Gibbs', 'Gibbs', 'Prince', 'Boucher', 'Kemp', 'Gibbs', 'Gilchrist who tumbles to his left but', 'Gibbs', 'Gibbs', 'Gibbs', 'Gibbs who is back into his crease', 'Gibbs', 'Gibbs', 'Kemp', 'Gibbs', 'Gibbs', 'McGrath McGrath to Gibbs', 'Gibbs', 'Gibbs', 'Kemp', 'Gibbs', 'Gibbs', 'Gibbs', 'Kemp', 'Kemp', 'Kemp', 'Gibbs', 'Kemp', 'Gibbs', 'Gibbs', 'Kemp', 'Gibbs', 'Gibbs', 'Gibbs', 'Gibbs', 'Gibbs', 'Kemp', 'Gibbs', 'Gibbs', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Gibbs', 'Gibbs', 'Gibbs', 'Gibbs', 'Gibbs', 'Kemp', 'Kemp', 'Kemp', 'Gibbs', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Gibbs', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Gibbs', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Kemp', 'Gibbs', 'Kemp', 'Gibbs', 'Gibbs', 'Kemp', 'Gibbs', 'Gibbs', 'Gibbs', 'Gibbs

In [86]:
def get_unique_player_names(player_names):
    # Convert the list to a set to remove duplicates, then convert back to a list
    unique_player_names = list(set(player_names))
    return unique_player_names

player_names = get_unique_player_names(player_names)
player_names

['Hogg at mid',
 'Gilchrist who tumbles to his left but',
 'Mr Hussey',
 'Hayden',
 'Boucher',
 'Australia',
 'Tait',
 'Gibbs',
 'Kallis Kallis to Ponting',
 'Smith at mid',
 'Gibbs at point Kallis to Ponting',
 'Gibbs who is back into his crease',
 'Hall',
 'Langeveldt',
 'McGrath McGrath to Gibbs',
 'Prince',
 'Symonds',
 'Hall who stick out his right hand and fields Kemp to Clarke',
 'Ponting',
 'Kemp',
 'Pollock',
 'Hayden who works it through midwicket Kallis to Clarke',
 'Nel',
 'Hayden in the deep',
 'Kallis',
 'Clarke',
 'Gilchrist',
 'Tait at long',
 'Smith']

In [87]:
s = ""
for  p in player_names :
    s+=p


In [105]:
def getplayers(document):
   
    return (ollama("give me the name of the players from this text  check properly if its a name add it "+document))
    
h =getplayers(s)

In [106]:
print(h)




Sure, here are the players mentioned in the text:

1. Hogg
2. Gilchrist
3. Ponting
4. Hussey
5. Hayden
6. Boucher
7. Tait
8. Gibbs
9. Kallis
10. Smith
11. Langeveldt
12. McGrath
13. Prince
14. Hall

Note that some of these players are mentioned multiple times in the text, but I have only listed them once in the order they appear.


In [107]:
player_pattern = r"\b[A-Z][a-zA-Z]+\b"

# Extract player names from the string
player_names = re.findall(player_pattern, h)

print(player_names)
player_names_1 = player_names[2:-1]
player_names_1

['Sure', 'Hogg', 'Gilchrist', 'Ponting', 'Hussey', 'Hayden', 'Boucher', 'Tait', 'Gibbs', 'Kallis', 'Smith', 'Langeveldt', 'McGrath', 'Prince', 'Hall', 'Note']


['Gilchrist',
 'Ponting',
 'Hussey',
 'Hayden',
 'Boucher',
 'Tait',
 'Gibbs',
 'Kallis',
 'Smith',
 'Langeveldt',
 'McGrath',
 'Prince',
 'Hall']

In [108]:
player_st =""
for player in player_names_1:
    player_st+=player
    player_st+=","
player_st

'Gilchrist,Ponting,Hussey,Hayden,Boucher,Tait,Gibbs,Kallis,Smith,Langeveldt,McGrath,Prince,Hall,'

In [124]:
from langchain.llms import Ollama
ollama = Ollama(base_url="http://localhost:11434",model="llama3")
print(ollama("is they sky blue"))

A classic question!

The answer is: It depends on who you ask and how you define "blue."

From Earth, the sky generally appears blue during the daytime because of a phenomenon called Rayleigh scattering. This is when shorter (blue) wavelengths of light are scattered more than longer (red) wavelengths by the tiny molecules of gases in the atmosphere, such as nitrogen and oxygen. This scattering effect gives the sky its blue hue.

However:

1. At sunrise and sunset, the sky can take on hues of orange, pink, and purple due to the scattering of longer wavelengths of light.
2. In areas with high levels of air pollution or dust, the sky may appear more hazy or gray.
3. From space, the Earth's atmosphere appears more like a thin layer of blue-gray, as the blue color is not as pronounced when viewed from outside the atmosphere.
4. On other planets, the sky can have different colors due to their unique atmospheric compositions and conditions. For example, Mars has a reddish sky due to iron oxid

In [127]:
print(ollama("give me all the things said about the australian team by the commentators  " + corpus))

Here are the comments made about the Australian team:

* "The defending champions" (referring to Australia's status as the reigning World Cup champions)
* "A clinical performance from the defending champions" (summarizing Australia's dominant display in the match)
* "They have done the business and will meet Sri Lanka on Saturday in a re-match of the 1996 final." (anticipating the upcoming semi-final between Australia and Sri Lanka)

There are no specific comments made about individual Australian players, apart from Glenn McGrath being referred to as the "Player of the match".


In [128]:
print(ollama("give me all the things said about the southafrican  team by the commentators  " + corpus))

Here are the comments made about the South African team:

* "dreadful" - used to describe the performance of the South Africans
* "South Africans acknowledge the support they have received and then head off to lick their wounds" - indicating that the South African team is disappointed with their loss.

Note that these comments are from the commentators, not from the players or coaches.


In [130]:
import spacy


nlp = spacy.load("en_core_web_sm")



doc = nlp(corpus)


country_names = [ent.text for ent in doc.ents if ent.label_ == "GPE"]


print("Country Names:", country_names[0:2])


Country Names: ['Australia', 'South Africa']


In [131]:
st_team1=ollama("give me all the things said about the"+ country_names[0] +"team by the commentators  " + corpus)
st_team2=ollama("give me all the things said about the"+ country_names[1] +"team by the commentators  " + corpus)



In [138]:
print(st_team2)

Here are the comments made about South Africa:

* "Dreadful" - referring to their performance in the game
* "Poor for neutrals, dreadful for South Africans" - summarizing the outcome of the match from a neutral's perspective and the disappointment of the South African fans.

These are the only specific comments made about South Africa by the commentators.


In [139]:
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer



Sentiment Score: {'neg': 0.184, 'neu': 0.816, 'pos': 0.0, 'compound': -0.9042}



In [140]:
def vader_Sent():
    analyzer = SentimentIntensityAnalyzer()

    sentiment_score = analyzer.polarity_scores(st_team2)
    
    print("Sentiment Score:", sentiment_score)
    print()


